server/src/modules/notifications/notifications-interfaces.ts

```diff
@@ -0,0 +1,103 @@
+export interface NotificationAgentConfig {
+  enabled: boolean;
+  types?: number[];
+  options: Record<string, unknown>;
+}
+export interface NotificationAgentDiscord extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    botUsername?: string;
+    botAvatarUrl?: string;
+    webhookUrl: string;
+  };
+}
+
+export interface NotificationAgentSlack extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    webhookUrl: string;
+  };
+}
+
+export interface NotificationAgentEmail extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    emailFrom: string;
+    smtpHost: string;
+    smtpPort: number;
+    secure: boolean;
+    ignoreTls: boolean;
+    requireTls: boolean;
+    authUser?: string;
+    authPass?: string;
+    allowSelfSigned: boolean;
+    senderName: string;
+    pgpPrivateKey?: string;
+    pgpPassword?: string;
+  };
+}
+
+export interface NotificationAgentLunaSea extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    webhookUrl: string;
+    profileName?: string;
+  };
+}
+
+export interface NotificationAgentTelegram extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    botUsername?: string;
+    botAPI: string;
+    chatId: string;
+    sendSilently: boolean;
+  };
+}
+
+export interface NotificationAgentPushbullet extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    accessToken: string;
+    channelTag?: string;
+  };
+}
+
+export interface NotificationAgentPushover extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    accessToken: string;
+    userToken: string;
+    sound: string;
+  };
+}
+
+export interface NotificationAgentWebhook extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    webhookUrl: string;
+    jsonPayload: string;
+    authHeader?: string;
+  };
+}
+
+export interface NotificationAgentGotify extends NotificationAgentConfig {
+  options: NotificationAgentConfig['options'] & {
+    url: string;
+    token: string;
+  };
+}
+
+export enum NotificationAgentKey {
+  DISCORD = 'discord',
+  EMAIL = 'email',
+  GOTIFY = 'gotify',
+  PUSHBULLET = 'pushbullet',
+  PUSHOVER = 'pushover',
+  SLACK = 'slack',
+  TELEGRAM = 'telegram',
+  WEBHOOK = 'webhook',
+  LUNASEA = 'lunasea',
+}
+
+export enum NotificationType {
+  MEDIA_ADDED_TO_COLLECTION = 2,
+  MEDIA_REMOVED_FROM_COLLECTION = 4,
+  MEDIA_ABOUT_TO_BE_HANDLED = 8,
+  MEDIA_HANDLED = 16,
+  RULE_HANDLING_FAILED = 32,
+  COLLECTION_HANDLING_FAILED = 64,
+  TEST_NOTIFICATION = 128,
+}
```

Comment:
This can be tackled in a future PR, but this could be moved to shared to save you having to call `/notifications/types`